docs-src/concepts/what-is-worker-versioning.md

```diff
@@ -0,0 +1,187 @@
+normal [deterministic constraints](/workflows/#deterministic-constraints), and you might need to use
+one of the [versioning APIs](/workflows/#workflow-versioning).
+
+The feature also permits you to make incompatible changes to Activity Definitions in conjunction
+with incompatible changes to Workflow Definitions that use those Activities. This works because any
+Activity scheduled by a Workfow on the same Task Queue will, by default, only be dispatched to
+Workers that have a compatible build ID with the Workflow that scheduled it. Thus, if you wish
+to change the type signature of an Activity Definition while creating a new incompatible build ID
+for a Worker, you can do so without concern that the Activity will fail to execute on some other
+Worker with an incompatible definition. The same principle applies to Child Workflows.
+
+Note that if you schedule an Activity or a Child Workflow on _different_ Task Queue from the one
+the Workflow is running on, they will be not be assigned any particular version.
+
+### How to use Worker Versioning
+
+To use this feature, follow these steps:
+
+1. Define Worker build-identifier version sets for the Task Queue.
+   You can use either the `temporal` CLI or your choice of SDK.
+   **_TODO: Links_**
+2. Enable the feature on your Worker by specifying a build identifier.
+   **_TODO: Links for each SDK's Worker options_**
+
+#### Defining the version sets
+
+Whether you use `temporal` cli or an SDK, updating the version sets feels the same. You specify the
+Task Queue that you're targeting, the build identifier that you're adding (or promoting), whether it
+becomes the new default version, and any existing versions it should be considered compatible with.
+
+The rest of this section uses updates to one Task Queue's version sets as examples.
+
+A Task Queue starts its life in an unversioned state. Workers who have not opted into the feature
+receive Tasks as usual. Workers who have opted in don't poll the Task Queue because they can't
+receive Tasks for Workflows that have no associated version.
+
+:::note Versions don't need to be semver!
+
+The versions in the following examples look like semver versions for clarity, but they don't need to
+be. Versions can be any arbitrary string.
+
+:::
+
+In the below examples, `(d)` means that set or version is the default.
+
+First, we add a version `1.0` to the Task Queue as the new default.
+Our version sets now look like this:
+
+| set 1 (d) |
+| --------- |
+| 1.0 (d)   |
+
+All new Workflows started on the Task Queue have their first Tasks assigned to version `1.0`.
+Workers with their build identifier set to `1.0` receive these Tasks.
+
+If Workflows that don't have an assigned version are still running on the Task Queue, Workers
+without a version take those tasks. So ensure that such Workers are still operational if any
+Workflows were open when you added the first version. If you deployed any Workers with a _different_
```

Comment:
Again, we need to explain that querying closed workflows also requires leaving the workers around.

Reply:
Addressed in the earlier spot.